### PR TITLE
chore: switch to @dimensiondev/metamask-extension-provider

### PR DESCRIPTION
**Pull request by guanbinrui**
ALERT: MetaMask <9.x is not compatiable.

Learn more: https://github.com/MetaMask/extension-provider/pull/17 https://github.com/DimensionDev/extension-provider/pull/2

### CODE REVIEW

File: packages/maskbook/src/web3/types.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ export enum CurrencyType {
 }
 
 export interface MetaMaskInpageProvider extends ReturnType<typeof createMetaMaskProvider> {
-    _metamask?: { isUnlocked: () => Promise<boolean> }
+    _metamask: { isUnlocked: () => Promise<boolean> }
```
Comment (guanbinrui) [inline packages/maskbook/src/web3/types.ts:14]: @zhouhanseng It's OK?

Comment (zhouhanseng) [reply packages/maskbook/src/web3/types.ts:14]: Sorry, I'm late. Yes now the `?` should be removed.
Before @metamask/inpage-provider@7.0.0 defines the type at `index.d.ts`, which does not include  `_metamask`.
Now @metamask/inpage-provider@8.0.3 defines the type at `dist/index.d.ts` which add `_metamask`,

**Review by zhouhanseng — APPROVED** [review]

File: packages/maskbook/src/web3/types.ts (hunk 1 above)
Comment (zhouhanseng) [reply packages/maskbook/src/web3/types.ts:14]: Sorry I'm late. Yes! `?` should be removed now.

Previously inpage-provider@7.0.0 defines types at `index.d.ts` which does not include `_metamask `. Now inpage-provider@8.0.3 defines types at `dist/index.d.ts` which include it.

**Comment by Tedko** [comment]
so we will require user to update MM to >= v9 right @guanbinrui

**Review by Tedko — APPROVED** [review]

**Comment by Tedko** [comment]
Will add a way to make it compatible with MM v<9 by @guanbinrui

**Pull request merged**